tests/pipelines/run_compiled_model_hotswap.py

```diff
@@ -0,0 +1,151 @@
+from peft import LoraConfig, get_peft_model_state_dict
+from peft.tuners.tuners_utils import BaseTunerLayer
+
+from diffusers import StableDiffusionPipeline, UNet2DConditionModel
+from diffusers.utils.testing_utils import floats_tensor
+
+
+torch_device = "cuda" if torch.cuda.is_available() else "cpu"
+
+
+def get_small_unet():
+    # from  UNet2DConditionModelTests
+    torch.manual_seed(0)
+    init_dict = {
+        "block_out_channels": (4, 8),
+        "norm_num_groups": 4,
+        "down_block_types": ("CrossAttnDownBlock2D", "DownBlock2D"),
+        "up_block_types": ("UpBlock2D", "CrossAttnUpBlock2D"),
+        "cross_attention_dim": 8,
+        "attention_head_dim": 2,
+        "out_channels": 4,
+        "in_channels": 4,
+        "layers_per_block": 1,
+        "sample_size": 16,
+    }
+    model = UNet2DConditionModel(**init_dict)
+    return model.to(torch_device)
+
+
+def get_unet_lora_config():
+    # from test_models_unet_2d_condition.py
+    rank = 4
+    unet_lora_config = LoraConfig(
+        r=rank,
+        lora_alpha=rank,
+        target_modules=["to_q", "to_k", "to_v", "to_out.0"],
+        init_lora_weights=False,
+        use_dora=False,
+    )
+    return unet_lora_config
+
+
+def get_dummy_input():
+    # from UNet2DConditionModelTests
+    batch_size = 4
+    num_channels = 4
+    sizes = (16, 16)
+
+    noise = floats_tensor((batch_size, num_channels) + sizes).to(torch_device)
+    time_step = torch.tensor([10]).to(torch_device)
+    encoder_hidden_states = floats_tensor((batch_size, 4, 8)).to(torch_device)
+
+    return {"sample": noise, "timestep": time_step, "encoder_hidden_states": encoder_hidden_states}
+
+
+def get_lora_state_dicts(modules_to_save):
+    state_dicts = {}
+    for module_name, module in modules_to_save.items():
+        if module is not None:
+            state_dicts[f"{module_name}_lora_layers"] = get_peft_model_state_dict(module)
+    return state_dicts
+
+
+def set_lora_device(model, adapter_names, device):
+    # copied from LoraBaseMixin.set_lora_device
+    for module in model.modules():
+        if isinstance(module, BaseTunerLayer):
+            for adapter_name in adapter_names:
+                module.lora_A[adapter_name].to(device)
+                module.lora_B[adapter_name].to(device)
+                # this is a param, not a module, so device placement is not in-place -> re-assign
+                if hasattr(module, "lora_magnitude_vector") and module.lora_magnitude_vector is not None:
+                    if adapter_name in module.lora_magnitude_vector:
+                        module.lora_magnitude_vector[adapter_name] = module.lora_magnitude_vector[adapter_name].to(
+                            device
+                        )
+
+
+def check_hotswap(do_hotswap):
+    dummy_input = get_dummy_input()
+    unet = get_small_unet()
+    lora_config = get_unet_lora_config()
+    unet.add_adapter(lora_config)
+    torch.manual_seed(42)
+    out_base = unet(**dummy_input)["sample"]
```

Comment:
`out_base` gives me the impression that the output is from the base original model whereas the model we're testing here has already being injected with an adapter config. So, `out_with_lora_config`?